### DISCUSSION TITLE
Oxidus | Palette color generator for Shadcn projects

### DISCUSSION THREAD

**Discussion by JairHuamanBellido** [shadcn-ui/ui#1884] [Show and tell]
Hello everyone,

We are excited to introduce our latest creation: Oxidus 🚀

Oxidus is your go-to tool for effortlessly creating and customizing color palettes based on shadcn. Oxidus offers a new level of color management:

🎨 Create Custom Color Palettes: Craft stunning palettes with ease, allowing you to visualize your designs like never before.

📤 Export & Import Themes: Seamlessly save and share your color themes with colleagues or the community, ensuring consistency in your projects.

🎨 Dynamic Color Modification: Adjust the primary color's lightness, darkness, saturation, and hue angle, providing you with ultimate control over your palette.

🌓 Dark and Light Modes: Experience Oxidus in the mode that suits your preference - whether it's the sleek Dark Mode for a professional touch or the classic Light Mode for a bright and airy design.

📋 Generate CSS Code: Instantly copy CSS code for the colors you've generated in HSL format, making integration into your projects.

Please let us know your thoughts, in order to improve our user experience or fix some bugs 🙏 

Github: https://github.com/JairHuamanBellido/Oxidus
Web:  https://oxidus.vercel.app/

<img width="1672" alt="image" src="https://github.com/shadcn-ui/ui/assets/38355200/9119997b-a28f-4d7f-8447-e3d347154aca">

<img width="1672" alt="image" src="https://github.com/shadcn-ui/ui/assets/38355200/bd8e7416-bf6f-4551-9331-b68d3e61bb60">

<img width="1672" alt="image" src="https://github.com/shadcn-ui/ui/assets/38355200/8132c234-333f-493f-aefb-fa9a767777dd">

<img width="1672" alt="image" src="https://github.com/shadcn-ui/ui/assets/38355200/50d3dedf-856f-4b1a-88ec-5c781f348e32">

<img width="1672" alt="image" src="https://github.com/shadcn-ui/ui/assets/38355200/de5ecc91-f5ac-49f1-a081-315970aa0438">

**Comment by KagamiChan** [comment]
I've used your site to generate a theme, and I think there could be some improvements. In official themes, the primary foreground (`--primary-foreground`) color would vary to ensure an enough contrast, for example:

official theme yellow as primary:
<img width="1041" alt="image" src="https://github.com/shadcn-ui/ui/assets/3816900/42bd616c-ea97-4299-8e0a-7a8a8caa046c">

the app with the same color:
<img width="502" alt="image" src="https://github.com/shadcn-ui/ui/assets/3816900/0f328841-a6fd-45e9-b3d4-38294e2323f1">


It would be great if we could calculate the contrast and switch the color

**Comment by freecolortool** [comment]
Great tool! The generated palettes are clean and easy to work with in Shadcn projects.

I think it would be interesting to see integration with a [Live Camera Color Picker](https://freecolortool.com/live-camera-color-extractor.html), allowing users to capture colors from real-world objects and instantly generate matching palettes. This could make creating custom themes even faster and more intuitive.

Thanks for building and sharing this project!

**Reply by michael-masarik** [reply]
💯

**Comment by michael-masarik** [comment]
Great tool! Thank you so much - it is so awesome to be able to preview what changes to each var does to the various components.

**Comment by freecolortool** [comment]
I've started using a[ Color Combiner ](https://freecolortool.com/online-color-mixer.html)much earlier in my workflow instead of waiting until the UI is almost finished. Testing multiple color combinations before writing CSS has helped me create more consistent interfaces and spend less time making visual tweaks later.

I mainly use it to compare button colors, backgrounds, accent colors, and hover states before implementing them. It makes it easier to build a cohesive design system and reduces the back-and-forth of changing HEX values throughout the project.

It's a small step, but it's made my UI workflow noticeably faster. I'm interested to hear if others follow a similar approach or have different methods for choosing color combinations.
